### ISSUE TITLE
Building from source uses 32 bit PCRE on 64 bit OS

### ISSUE THREAD

**Issue by yerlaser** [fish-shell/fish-shell#7690]
I have compiled Fish from sources on a 64 bit Ubuntu server.
During compilation I notice a message saying: 'using system pcre'
When I install this build of Fish to a 64 bit Kubuntu machine and run it, I get an error saying that libpcre2-32-0 is not found.
And indeed after installing libpcre2-32-0 package it runs successfully.
However, if I read correctly this library is 32-bit.
Why does Fish depend on a 32-bit library on a 64-bit OS?

I also tried to compile in portable mode adding these lines to cmake config:

SET(CMAKE_FIND_LIBRARY_SUFFIXES ".a")
SET(BUILD_SHARED_LIBS OFF)
SET(CMAKE_EXE_LINKER_FLAGS "-static")

${CURSES_LIBRARY} termcap ${CURSES_EXTRA_LIBRARY} "-Wl,--whole-archive -lpthread -Wl,--no-whole-archive" ${CMAKE_DL_LIBS}

This compiles and I can run Fish on Kubuntu, however, whenever I type tilde sign (~) anywhere on the command line it immediately crashes with Core Dump.

**Comment by faho** [comment]
This is the correct package. It's not about being i686 or amd64, it's about having 32-bit wide *characters*. Something like `file /usr/lib64/libpcre2-32.so.0.10.1` will confirm that it is indeed compiled for x86-64.

>I also tried to compile in portable mode adding these lines to cmake config:

This is not supported.

**Comment by zanchey** [comment]
> This compiles and I can run Fish on Kubuntu, however, whenever I type tilde sign (~) anywhere on the command line it immediately crashes with Core Dump.

See https://stackoverflow.com/questions/15165306/compile-a-static-binary-which-code-there-a-function-gethostbyname, https://stackoverflow.com/questions/57476533/why-is-statically-linking-glibc-discouraged for why this is failing. There is probably a warning when linking fish if you do a verbose build.

**Comment by yerlaser** [comment]
Hi @faho and @zanchey,

Thanks for pointing me in the right direction.
After reviewing the cmake config I found that it references cmake/PCRE2.cmake where it decides whether to use system or bundled library. I altered it to force bundled and now Fish runs in Kubuntu even without installing the said package.
This doesn't make Fish fully static, but at least more portable.

I understand, that this may not be optimal for general use, but in my use case I want my Kubuntu installation as vanilla as possible and so this workaround suits me perfectly.

Thanks and have a nice day!
